frontend/src/components/molecules/ChartCards/StableCoinTokenDistributionByHolder.vue

```diff
@@ -28,20 +29,17 @@ const distributionValues = ref<
 >([])
 
 watchEffect(() => {
-	const holders = props.tokenTransferData?.stablecoin?.holdings || []
-
-	if (holders.length === 0) {
-		distributionValues.value = []
-		return
-	}
-
-	setTimeout(() => {
-		distributionValues.value = holders.map((ele: HoldingResponse) => ({
-			address: shortenHash(ele.address),
-			symbol: ele.assetName || '',
-			percentage: (ele.percentage ?? 0).toFixed(2),
-		}))
-		isLoading.value = false
-	}, 1000)
+	distributionValues.value = []
+	props.tokenTransferData?.map(item => {
+		const address = item.accountAddress.asString
+		const amount = BigInt(item.amount.value)
+		const percentage = (Number(amount) / Number(props.totalSupply ?? 0n)) * 100
+		const symbol = item.tokenId ?? ''
```

Comment:
Not sure when it would make sense to let `tokenId` be undefined and then parse it into a dummy value here.